include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.h

```diff
@@ -654,9 +655,23 @@ class SpatialOperatorBase : public dealii::Subscriptor
   mutable VectorType const * pressure_ptr;
 
   /*
-   * LES turbulence modeling.
-   */
-  TurbulenceModel<dim, Number> turbulence_model;
+   * LES turbulence and generalized Newtonian models with templated activation.
+   */
+  ViscosityModel<dim,
+                 Number,
+                 true /*use_turbulence_model*/,
+                 false /*use_generalized_newtonian_model*/>
+    turbulent_viscosity_model;
+  ViscosityModel<dim,
+                 Number,
+                 true /*use_turbulence_model*/,
+                 true /*use_generalized_newtonian_model*/>
+    generalized_newtonian_turbulent_viscosity_model;
+  ViscosityModel<dim,
+                 Number,
+                 false /*use_turbulence_model*/,
+                 true /*use_generalized_newtonian_model*/>
+    generalized_newtonian_viscosity_model;
```

Comment:
This does not look like a modular design that can be extended in the future. This is something I would like to discuss in general. Maybe an issue is needed for this.